test/TestApps/RawNotificationTriggerApp/RawNotificationTriggerApp.cpp

```diff
@@ -0,0 +1,160 @@
+﻿// MessageTrigger.cpp : This file contains the 'main' function. Program execution begins and ends there.
+//
+
+#include "pch.h"
+#include <string>
+#include <iostream>
+#include <sstream>
+#include <winrt/Windows.Networking.PushNotifications.h>
+#include <winrt/Windows.Foundation.h>
+#include <winrt/Windows.Storage.Streams.h>
+#include <winrt/Windows.Web.Http.Headers.h>
+#include <Windows.h>
+#include <sddl.h>
+
+using namespace WEX::Common;
+using namespace WEX::Logging;
+using namespace WEX::TestExecution;
+
+using namespace winrt;
+using namespace winrt::Windows::Storage;
+using namespace winrt::Windows::Storage::Streams;
+using namespace winrt::Windows::Foundation;
+using namespace Windows::Web::Http;
+
+bool g_quit = false;
+std::wstring g_channelUri = L"";
+std::wstring g_payload = L"";
+
+void cleanInputBuffer()
+{
+    std::string trash;
+    std::getline(std::cin, trash, '\n');
+}
+
+std::wstring BuildNotificationPayload(bool isToast)
+{
+    std::wstring x_wns_type = isToast ? L"\"wns/toast\"" : L"\"wns/raw\"";
+    std::wstring contentType = isToast ? L"\"text/xml\"" : L"\"application/octet-stream\"";
+    if (g_channelUri.empty())
+    {
+        std::cout << "Please set the channel uri before sending notification." << std::endl << std::endl;
+        return L"";
+    }
+    else if (g_channelUri.empty())
+    {
+        std::cout << "Please set the payload before sending notification." << std::endl << std::endl;
+        return L"";
+    }
+    
+    std::wstring channelUri = L"\"ChannelUri\":\"" + g_channelUri + L"\"";
+    std::wstring x_wns_typeJson = L"\"X_WNS_Type\": " + x_wns_type;
+    std::wstring contentTypeJson = L"\"Content_Type\": " + contentType;
+    std::wstring payload = L"\"Payload\": \"" + g_payload + L"\"";
+    std::wstring delay = L"\"Delay\": \"false\"";
+    return { L"{" + channelUri + L"," + x_wns_typeJson + L"," + contentTypeJson + L"," + payload + L"," + delay + L"}" };
+}
+
+void SendPushNotification(bool isToast) {
+    cleanInputBuffer();
+    HttpResponseMessage httpResponseMessage;
+    std::wstring httpResponseBody;
+    std::wstring notificationPayload = BuildNotificationPayload(isToast);
+    if (notificationPayload.empty())
+    {
+        return;
+    }
+
+    try {
+        // Construct the HttpClient and Uri. This endpoint is for test purposes only.
+        HttpClient httpClient;
+        Uri requestUri{ L"http://localhost:7071/api/PostPushNotification" };
+
+        // Construct the JSON to post.
+        HttpStringContent jsonContent(
+            notificationPayload,
+            UnicodeEncoding::Utf8,
+            L"application/json");
+
+        // Post the JSON, and wait for a response.
+        httpResponseMessage = httpClient.PostAsync(
+            requestUri,
```

Comment:
Just a little concerned that we have no auth for the endpoint uri here. Is this going to be fixed later?

Reply:
Auth is being performed on the TestAppServer side as of now.